### PR TITLE
Fix TabClose autocommand via close_windows

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/neovim/neovim/issues/7781

### CODE REVIEW

File: src/nvim/window.c [hunk 1]
```diff
@@ -1741,6 +1741,11 @@ void close_windows(buf_T *buf, int keep_curwin)
     } else
       wp = wp->w_next;
   }
+  if (count != tabpage_index(NULL)) {
+    char_u prev_idx[NUMBUFLEN];
+    sprintf((char *)prev_idx, "%i", tabpage_index(curtab));
```
Comment (justinmk) [inline src/nvim/window.c]: Should use snprintf. Also might as well check `has_event(EVENT_TABCLOSED)`, no?

**Comment by justinmk** [comment]
Mind adding a test?

File: src/nvim/window.c (hunk 1 above)
Comment (blueyed) [reply src/nvim/window.c]: This was copied/moved IIRC.
Changed it now, with tests added.

**Comment by blueyed** [comment]
Forwarded to Vim: https://github.com/vim/vim/pull/2510.

File: src/nvim/window.c [hunk 2]
```diff
@@ -2121,6 +2120,10 @@ void win_close_othertab(win_T *win, int free_buf, tabpage_T *tp)
       ptp->tp_next = tp->tp_next;
     }
     free_tp = TRUE;
+
+    if (has_event(EVENT_TABCLOSED)) {
+        apply_autocmds(EVENT_TABCLOSED, prev_idx, prev_idx, false, win->w_buffer);
```
Comment (justinmk) [inline src/nvim/window.c]: Should be 2-space indent, I fixed this in the merge. Same for the test file.

**Comment by justinmk** [comment]
Thanks for the extra test coverage, let's merge it and see what happens :)

**Pull request merged**